Composer/packages/client/src/components/Notifications/NotificationCard.tsx

```diff
@@ -132,21 +157,59 @@ export type NotificationProps = {
   onHide?: (id: string) => void;
 };
 
+const makeLinkLabel = (link: NotificationLink) => (
+  <ActionButton styles={linkButton} onClick={link.onClick}>
+    {link.label}
+  </ActionButton>
+);
+
 const defaultCardContentRenderer = (props: CardProps) => {
-  const { title, description, type, link } = props;
+  const { title, description, type, link, links, leftLinks, rightLinks } = props;
+
+  const rightLinkList = rightLinks ?? links ?? [link];
+  const leftLinkList = leftLinks ?? [];
+
+  const stackProps: IStackProps = {
+    horizontal: true,
+    horizontalAlign: 'space-between',
+    tokens: {
+      childrenGap: '20px',
+      padding: '0 16px 0 0',
+      maxHeight: '24px',
+    },
+  };
+
   return (
     <div css={cardContent}>
       {type === 'error' && <Icon css={errorType} iconName="ErrorBadge" />}
       {type === 'success' && <Icon css={successType} iconName="Completed" />}
       {type === 'warning' && <Icon css={warningType} iconName="Warning" />}
+      {type === 'question' && <Icon css={questionType} iconName="UnknownSolid" />}
+      {type === 'congratulation' && <Icon css={congratulationType} iconName="Trophy2Solid" />}
+      {type === 'custom' && (
+        <Icon
+          css={css`
+            margin-top: ${iconMargin};
+            color: ${props.color ?? SharedColors.gray10};
+          `}
+          iconName={props.icon ?? 'UnknownSolid'}
+        />
+      )}
       <div css={cardDetail}>
         <div css={cardTitle}>{title}</div>
         {description && <div css={cardDescription}>{description}</div>}
-        {link && (
-          <ActionButton css={linkButton} onClick={link.onClick}>
-            {link.label}
-          </ActionButton>
-        )}
+        <Stack horizontal horizontalAlign="space-between">
+          <Stack {...stackProps}>
+            {leftLinkList.map(
+              (link) => link != null && <Stack.Item key={link.label}>{makeLinkLabel(link)}</Stack.Item>
```

Comment:
When will `link` be `null`?

Reply:
I'm not sure either, but it wouldn't typecheck without this and it's a harmless thing to check anyway.

Reply:
The type checker was probably complaining because you previously added `null` and `undefined` to the link properties in the `Notification` type in Composer/packages/types/src/shell.ts. If you restart your Typescript Server, the type warning would probably resolve.